test/y2storage/filesystems/type_test.rb

```diff
@@ -105,4 +107,80 @@
       end
     end
   end
+
+  describe "#iocharset and #codepage" do
+    it "return the correct values in a utf8 locale" do
+      Yast::Encoding.SetUtf8Lang(true)
+      Yast::Encoding.SetEncLang("cs_CZ")
+      expect(described_class::VFAT.iocharset).to eq "utf8"
+      expect(described_class::VFAT.codepage).to eq "437"
+    end
+
+    it "return the correct values in a legacy cs_CZ locale" do
+      Yast::Encoding.SetUtf8Lang(false)
+      Yast::Encoding.SetEncLang("cs_CZ")
+
+      expect(described_class::VFAT.iocharset).to eq "iso8859-2"
+      expect(described_class::VFAT.codepage).to eq "852"
+    end
+
+    it "return the correct values in a legacy de_DE locale" do
+      Yast::Encoding.SetUtf8Lang(false)
+      Yast::Encoding.SetEncLang("de_DE")
+      expect(described_class::VFAT.iocharset).to eq "iso8859-15"
+      expect(described_class::VFAT.codepage).to eq "437"
+    end
+
+    it "return the correct values in a utf8 ja_JP locale" do
+      Yast::Encoding.SetUtf8Lang(true)
+      Yast::Encoding.SetEncLang("ja_JP")
+      expect(described_class::VFAT.iocharset).to eq "utf8"
+      expect(described_class::VFAT.codepage).to eq "932"
+    end
+  end
+
+  describe "#default_fstab_options" do
```

Comment:
Nitpicking: I would shuffle the organization of contexts and examples a little bit, so the unit tests really check that the locale-independent types are indeed not affected by the locale (and that it stays like that in the future).

Something in this direction:

```ruby
RSpec.shared_examples "locale-independent filesystem types" do
  it "returns the correct options for ext2" do
    expect(described_class::EXT2.default_fstab_options).to eq ["acl", "user_xattr"]
  end

  it "includes no encoding options for ext2" do
    options = described_class::EXT2.default_fstab_options
    expect(options).to_not include(a_string_starting_with("iocharset"))
    expect(options).to_not include(a_string_starting_with("codepage"))
  end

  it "returns the correct options for ext3" do
    expect(described_class::EXT3.default_fstab_options).to eq  ["data=ordered", "acl", "user_xattr"]
  end

  it "includes no encoding options for ext3" do
    options = described_class::EXT3.default_fstab_options
    expect(options).to_not include(a_string_starting_with("iocharset"))
    expect(options).to_not include(a_string_starting_with("codepage"))
  end
end

context "in a utf8 locale" do
  before do
    Yast::Encoding.SetUtf8Lang(true)
    Yast::Encoding.SetEncLang("de_DE")
  end

  include_examples "locale-independent filesystem types" 

  it "returns the correct options for vfat" do
    expect(described_class::VFAT.default_fstab_options).to eq ["utf8=true"]
  end
end

context "in a non-utf8 cs_CZ locale" do
  before do
    Yast::Encoding.SetUtf8Lang(false)
    Yast::Encoding.SetEncLang("cs_CZ")
  end

  include_examples "locale-independent filesystem types" 

  it "returns the correct options for vfat" do
    expect(described_class::VFAT.default_fstab_options).to eq ["iocharset=iso8859-2", "codepage=852"]
  end
end
```

In fact, even more combinations can be achieved with less code, but that's a different topic. My main point is that the way to probe extX are locale-independent is actually testing then with several locales as it's done with the locale-dependant filesystems.

Reply:
OK, I'll see about that, but in the next PBI; I'd like to get this fix shipped with the upcoming next Beta by the end of tomorrow.